lib/setupHooks/inheritCargoArtifactsHook.sh

```diff
@@ -17,6 +17,12 @@ inheritCargoArtifacts() {
 
   mkdir -p "${cargoTargetDir}"
   if [ -f "${preparedArtifacts}" ]; then
+
+    local prevArtifactsCandidate="${preparedArtifacts}.prev"
+    if [ -f "${prevArtifactsCandidate}" ]; then
+      inheritCargoArtifacts "${prevArtifactsCandidate}" "$cargoTargetDir"
```

Comment:
In my change I did `readlink` here, as there are some checks in `inheritCargoArtifacts` that I'm not sure are going to be happy with a symlink. Even if it is working now, it will be easy to introduce (possibly silent, as it will only trigger rebuilds) bug in the future.

Reply:
Could you elaborate on what specifically wouldn't be happy with a symlink?

`inheritCargoArtifacts` already does some introspection on whether the input is a directory or not (which should work fine with or without symlinks) so I can't think of a situation where things might fail

Reply:
Nothing specific. Just a normal path passes a normal path there, not a symlink. I'm wary of symlinks vs non-symlinks issue. Some tools sometimes do some different things because of them.

But it will probably be OK. Nothing specific.

Reply:
I think this is fine for now (initial testing shows that the symlink is working out fine) so I'll mark this as resolved but happy to revisit if we ever find problems with it